docs/src/explanations/chisel-enum.md

```diff
@@ -0,0 +1,129 @@
+# ChiselEnum
+
+The ChiselEnum type is an easy way to get away from value encodings and reduce the chance of error when encoding muxes, opcodes, and functional unit operations. As opposed to regular Enums, ChiselEnums allow for function names to be explicitly passed through Bundles like IO. 
```

Comment:
Thanks for writing this!

what is a "regular Enum"? Do you mean `scala.Enum`, or something else?

Also, not sure what you mean by "allow for function names". What is a "function name"?

Reply:
By regular Enum I mean the Chisel enum. I can clarify that. 
By function name I mean like a opcode being passed into an ALU or CMP.